TriblerGUI/widgets/trustdisplaypage.py

```diff
@@ -0,0 +1,161 @@
+        """
+        focus_node = information["focus_node"]
+
+        grouped_edges = TrustDisplayPage.group_elements(information["edges"], "from")
+
+        # TODO: Do the procedure below for each of the nodes in the graph, whilst erasing duplicates.
+        focus_node_edges = TrustDisplayPage.get_combined_edges(grouped_edges, focus_node)
+
+        # TODO: Send the data to the display to display the data
+        return {"nodes": information["nodes"], "edges": focus_node_edges}
+
+    @staticmethod
+    def get_combined_edges(grouped_edges, node_name):
+        """
+        Combine the directed edges between the given node and other nodes to one edge per pair.
+
+        The attributes of the combined edges are calculated as follows:
+            - from: node_name
+            - to: to attribute from outgoing edge from node_name
+            - amount_up: amount from the outgoing edge from node_name
+            - amount_down: amount from the ingoing edge to node_name if any
+            - ratio: amount_up / (amount_up + amount_down)
+            - log_ratio: log(amount_up + 1) / (log(amount_up + 1) + log(amount_down + 1))
+
+        :param grouped_edges: the dictionary of edges, grouped by "from" attribute
+        :param node_name: the node name from which viewpoint each combine edge is created
+        :return: a list of combined edges with the described attributes
+        """
+        combined_edges = []
+        for edge in grouped_edges[node_name]:
+            inverse_edge = next((inv for inv in grouped_edges[edge["to"]] if inv["to"] == node_name), None)
+            up = edge["amount"]
+            down = 0
+            ratio = 0
+            log_ratio = 0
+            if inverse_edge:
+                down = inverse_edge["amount"]
+            if not(up == 0 and down == 0):
+                ratio = float(up) / (up + down)
+                log_ratio = log(up + 1) / (log(up + 1) + log(down + 1))
+            combined_edges.append({"from": node_name, "to": edge["to"], "amount_up": up, "amount_down": down,
+                                   "ratio": ratio, "log_ratio": log_ratio})
+        return combined_edges
+
+    @staticmethod
+    def group_elements(elements, attribute):
+        """
+        Group elements in the list by the given attribute.
+                
+        Use a defaultdict to group the elements by their attribute. This function returns a dictionary with keys
+        corresponding to the different occurring attributes.
+
+        :param elements: the list of elements which has to be grouped
+        :param attribute: the attribute from the elements which the elements must be grouped on
+        :return: a defaultdict with elements grouped by attribute name 
+        """
+        return_dict = defaultdict(list)
+        for element in elements:
+            return_dict[element[attribute]].append(element)
+
+        return return_dict
+
+    def get_focus_node(self):
```

Comment:
Please don't attempt to get Java-like type safety by using getters and setters with asserts. 

I would propose to not make `focus_node` protected and remove those asserts. We should write unittests to verify functionality, rather than programming defensively using asserts.